experimental/Schemes/Resolution_structure.jl

```diff
@@ -241,6 +380,45 @@ function _desing_curve(X::AbsCoveredScheme, I_sl::AbsIdealSheaf)
   return phi
 end
 
+function _desing_lipman(X::AbsCoveredScheme, I_sl::AbsIdealSheaf)
+  dim(X) == 2 || error("Lipman's algorithm is not applicable")
+
+  if dim(I_sl) == 1
+    Xnorm, phi = normalization(X)
+    incs = phi.inclusions
+    f = initialize_mixed_blow_up_sequence(phi,I_sl)
+    f_initialized = true
+  end    
+  
+  I_sl_temp = ideal_sheaf_of_singular_locus(Xnorm)
+  decomp = maximal_associated_points(I_sl_temp)
+
+  while !is_one(I_sl_temp)
+    while length(decomp) > 0
+      I = small_generating_set(pop!(decomp))
+      if !f_initialized
+        phi = blow_up(I)
+        f = initialize_mixed_blow_up_sequence(phi)::MixedBlowUpSequence
+        f_initialized = true
+      else
+        f = _do_blow_up!(f,I)
+        if length(decomp)>0 
+          decomp = [strict_transform(last_map(f),J) for J in decomp]
+        end
+      end
+    end
+    I_sl_temp = ideal_sheaf_of_singular_locus(domain(last_map(f)))
```

Comment:
You can add the exceptional locus here to make sure you don't get duplicate components which are already coming up from `strict_transform` above. Eventually, we should think about providing some extra argument to specify that `ideal_sheaf_of_singular_locus` is supposed to only look for singularities along the support of another ideal sheaf.

Reply:
The preceding loop exhausted all components of the singular locus as centers of blow-ups, so there is no risk of getting duplicates.